backend/main.py

```
@@ -9,46 +1,343 @@
+@app.get("/documents/stats", response_model=DocumentStats)
+async def get_document_stats() -> DocumentStats:
+    """Get statistics about all discovered documents."""
+    if not AUTO_RAG_AVAILABLE or not rag_system:
+        raise HTTPException(status_code=503, detail="RAG system not available")
+
+    try:
+        stats = rag_system.get_document_stats()
+        return DocumentStats(**stats)
+    except Exception as e:
+        logger.error(f"❌ Failed to get document stats: {e}")
+        raise HTTPException(status_code=500, detail=f"Failed to retrieve document statistics: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)